### PR TITLE
Extend gui page change button functionality

### PR DESCRIPTION
**Pull request by NyCodeGHG**
Added:
* GuiButtonPageChangeAdvanced (Like the normal one, but with ability to cancel the page change)
* Builder Functions for the above mentioned functionality

### CODE REVIEW

**Comment by bluefireoly** [comment]
I would add this functionality to the normal gui button instead of adding a completely new kind of button. Dou you see any problems with that approach?

**Comment by NyCodeGHG** [comment]
No, that would be great

**Comment by bluefireoly** [comment]
can you add this feature to the default pageChange button? if so, I would definitely merge this PR

**Comment by NyCodeGHG** [comment]
I will do this as soon as I have found the time.

**Review by bluefireoly — CHANGES_REQUESTED** [review]
you have to update the builder function as well (does this even compile?)

File: src/main/kotlin/net/axay/kspigot/gui/elements/GUIButtonPageChange.kt [hunk 1]
```diff
@@ -6,7 +6,8 @@ import org.bukkit.inventory.ItemStack
 class GUIButtonPageChange<T : ForInventory>(
     icon: ItemStack,
     calculator: GUIPageChangeCalculator,
-    onChange: ((GUIClickEvent<T>) -> Unit)?
+    beforeChange: ((GUIClickEvent<T>) -> Boolean)?,
+    afterChange: ((GUIClickEvent<T>) -> Unit)?
```
Comment (bluefireoly) [inline src/main/kotlin/net/axay/kspigot/gui/elements/GUIButtonPageChange.kt:10]: rename to `shouldChange` and `onChange`

**Review by bluefireoly — CHANGES_REQUESTED** [review]

**Comment by NyCodeGHG** [comment]
> you have to update the builder function as well (does this even compile?)

right, there was something

**Comment by NyCodeGHG** [comment]
Should I create new functions so that the api does not break?

**Comment by bluefireoly** [comment]
yes, you can create one new function and mark the old one as deprecated, and don't forget to add the `ReplaceWith()` parameter to the deprecated annotation

**Review by bluefireoly — APPROVED** [review]
Thank you!

**Pull request merged**